components/check-list/check-list-content.component.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
+ */
+
+import { DecimalPipe } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  computed,
+  input,
+  output,
+  signal,
+  TemplateRef,
+  ViewEncapsulation
+} from '@angular/core';
+import { FormsModule } from '@angular/forms';
+
+import { NzButtonModule } from 'ng-zorro-antd/button';
+import { NzCheckboxModule } from 'ng-zorro-antd/checkbox';
+import { NzOutletModule } from 'ng-zorro-antd/core/outlet';
+import { NzCheckListI18nInterface } from 'ng-zorro-antd/i18n';
+import { NzIconModule } from 'ng-zorro-antd/icon';
+import { NzProgressModule } from 'ng-zorro-antd/progress';
+
+import { NzItemProps } from './typings';
+
+@Component({
+  selector: 'nz-check-list-content',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  imports: [NzIconModule, NzProgressModule, NzOutletModule, NzCheckboxModule, NzButtonModule, FormsModule, DecimalPipe],
+  template: `
+    @let i18n = locale();
+    @if (visible()) {
+      @if (progressPercent() === 100) {
+        <div class="ant-check-list-header-finish">
+          <nz-icon nzType="check-circle" nzTheme="outline" class="ant-check-list-header-finish-icon" />
+          <h3 class="ant-check-list-header-finish-title">{{ i18n.checkListFinish }}</h3>
+          <button nz-button nzType="primary" style="margin: 24px" (click)="closePopover.emit(false)">
```

Comment:
Remove inline style (CSP compliant)

```suggestion
          <button nz-button nzType="primary" [style.margin.px]="24" (click)="closePopover.emit(false)">
```